The job **`dist-loongarch64-musl`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/18923684894/job/54025955754) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/54025955754) [(plain)](https://github.com/rust-lang/rust/commit/afcab8399ef11ca67e57b93217e92b1e76af0c2e/checks/54025955754/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

error[E0659]: `STATX_NLINK` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:79:26
    |
 79 |         const NLINK = c::STATX_NLINK;
    |                          ^^^^^^^^^^^ ambiguous name
    |
    = note: ambiguous because of multiple glob imports of a name in the same module
note: `STATX_NLINK` could refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:7:16
    |
---
512 | pub(crate) use statx_flags::*;
    |                ^^^^^^^^^^^^^^
    = help: consider adding an explicit import of `STATX_NLINK` to disambiguate

error[E0659]: `STATX_UID` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:82:24
    |
 82 |         const UID = c::STATX_UID;
    |                        ^^^^^^^^^ ambiguous name
    |
    = note: ambiguous because of multiple glob imports of a name in the same module
note: `STATX_UID` could refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:7:16
    |
---
512 | pub(crate) use statx_flags::*;
    |                ^^^^^^^^^^^^^^
    = help: consider adding an explicit import of `STATX_UID` to disambiguate

error[E0659]: `STATX_GID` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:85:24
    |
 85 |         const GID = c::STATX_GID;
    |                        ^^^^^^^^^ ambiguous name
    |
    = note: ambiguous because of multiple glob imports of a name in the same module
note: `STATX_GID` could refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:7:16
    |
---

error[E0659]: `STATX_ATIME` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:88:26
    |
 88 |         const ATIME = c::STATX_ATIME;
    |                          ^^^^^^^^^^^ ambiguous name
    |
    = note: ambiguous because of multiple glob imports of a name in the same module
note: `STATX_ATIME` could refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:7:16
    |
---

error[E0659]: `STATX_MTIME` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:91:26
    |
 91 |         const MTIME = c::STATX_MTIME;
    |                          ^^^^^^^^^^^ ambiguous name
    |
    = note: ambiguous because of multiple glob imports of a name in the same module
note: `STATX_MTIME` could refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:7:16
    |
---
512 | pub(crate) use statx_flags::*;
    |                ^^^^^^^^^^^^^^
    = help: consider adding an explicit import of `STATX_CTIME` to disambiguate

error[E0659]: `STATX_INO` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:97:24
    |
 97 |         const INO = c::STATX_INO;
    |                        ^^^^^^^^^ ambiguous name
    |
    = note: ambiguous because of multiple glob imports of a name in the same module
note: `STATX_INO` could refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:7:16
    |
---

error[E0659]: `STATX_BTIME` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:109:26
    |
109 |         const BTIME = c::STATX_BTIME;
    |                          ^^^^^^^^^^^ ambiguous name
    |
    = note: ambiguous because of multiple glob imports of a name in the same module
note: `STATX_BTIME` could refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:7:16
    |
---
512 | pub(crate) use statx_flags::*;
    |                ^^^^^^^^^^^^^^
    = help: consider adding an explicit import of `STATX_BTIME` to disambiguate

error[E0659]: `STATX_MNT_ID` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:112:27
    |
112 |         const MNT_ID = c::STATX_MNT_ID;
    |                           ^^^^^^^^^^^^ ambiguous name
    |
    = note: ambiguous because of multiple glob imports of a name in the same module
note: `STATX_MNT_ID` could refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:7:16
    |
---
512 | pub(crate) use statx_flags::*;
    |                ^^^^^^^^^^^^^^
    = help: consider adding an explicit import of `STATX_MNT_ID` to disambiguate

error[E0659]: `STATX_DIOALIGN` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:115:29
    |
115 |         const DIOALIGN = c::STATX_DIOALIGN;
    |                             ^^^^^^^^^^^^^^ ambiguous name
    |
    = note: ambiguous because of multiple glob imports of a name in the same module
note: `STATX_DIOALIGN` could refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:7:16
    |
  7 | pub(crate) use libc::*;
    |                ^^^^^^^
    = help: consider adding an explicit import of `STATX_DIOALIGN` to disambiguate
note: `STATX_DIOALIGN` could also refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:512:16
    |
512 | pub(crate) use statx_flags::*;
    |                ^^^^^^^^^^^^^^
    = help: consider adding an explicit import of `STATX_DIOALIGN` to disambiguate

error[E0659]: `STATX_ALL` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:118:24
    |
118 |         const ALL = c::STATX_ALL;
---
512 | pub(crate) use statx_flags::*;
    |                ^^^^^^^^^^^^^^
    = help: consider adding an explicit import of `STATX_ATTR_APPEND` to disambiguate

error[E0659]: `STATX_ATTR_NODUMP` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:140:27
    |
140 |         const NODUMP = c::STATX_ATTR_NODUMP as u64;
    |                           ^^^^^^^^^^^^^^^^^ ambiguous name
    |
    = note: ambiguous because of multiple glob imports of a name in the same module
note: `STATX_ATTR_NODUMP` could refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:7:16
    |
  7 | pub(crate) use libc::*;
    |                ^^^^^^^
    = help: consider adding an explicit import of `STATX_ATTR_NODUMP` to disambiguate
note: `STATX_ATTR_NODUMP` could also refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:512:16
    |
512 | pub(crate) use statx_flags::*;
    |                ^^^^^^^^^^^^^^
    = help: consider adding an explicit import of `STATX_ATTR_NODUMP` to disambiguate

error[E0659]: `STATX_ATTR_ENCRYPTED` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:143:30
    |
143 |         const ENCRYPTED = c::STATX_ATTR_ENCRYPTED as u64;
---
512 | pub(crate) use statx_flags::*;
    |                ^^^^^^^^^^^^^^
    = help: consider adding an explicit import of `STATX_ATTR_VERITY` to disambiguate

error[E0659]: `STATX_ATTR_DAX` is ambiguous
   --> /rust/deps/rustix-1.0.8/src/fs/statx.rs:155:24
    |
155 |         const DAX = c::STATX_ATTR_DAX as u64;
    |                        ^^^^^^^^^^^^^^ ambiguous name
    |
    = note: ambiguous because of multiple glob imports of a name in the same module
note: `STATX_ATTR_DAX` could refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:7:16
    |
  7 | pub(crate) use libc::*;
    |                ^^^^^^^
    = help: consider adding an explicit import of `STATX_ATTR_DAX` to disambiguate
note: `STATX_ATTR_DAX` could also refer to the constant imported here
   --> /rust/deps/rustix-1.0.8/src/backend/libc/c.rs:512:16
    |
512 | pub(crate) use statx_flags::*;
    |                ^^^^^^^^^^^^^^
    = help: consider adding an explicit import of `STATX_ATTR_DAX` to disambiguate

[RUSTC-TIMING] thin_vec test:false 0.480
   Compiling rustc-hash v1.1.0
[RUSTC-TIMING] rustc_hash test:false 0.071
   Compiling itoa v1.0.15
```

</details>

--- a/compiler/rustc_session/Cargo.toml
+++ b/compiler/rustc_session/Cargo.toml
@@ -27,10 +27,8 @@ tracing = "0.1"
 # tidy-alphabetical-end
 
 [target.'cfg(unix)'.dependencies]
-# FIXME: Remove this pin once this rustix issue is resolved
-# https://github.com/bytecodealliance/rustix/issues/1496
 # tidy-alphabetical-start
-libc = "=0.2.174"
+libc = "0.2"
 # tidy-alphabetical-end
 
 [target.'cfg(windows)'.dependencies.windows]

--- a/src/tools/rustbook/Cargo.toml
+++ b/src/tools/rustbook/Cargo.toml
@@ -10,9 +10,7 @@ edition = "2021"
 [dependencies]
 clap = "4.0.32"
 env_logger = "0.11"
-# FIXME: Remove this pin once this rustix issue is resolved
-# https://github.com/bytecodealliance/rustix/issues/1496
-libc = "=0.2.174"
+libc = "0.2"
 mdbook-trpl = { path = "../../doc/book/packages/mdbook-trpl" }
 mdbook-i18n-helpers = "0.3.3"
 mdbook-spec = { path = "../../doc/reference/mdbook-spec" }

